test/default_style.jl

```diff
@@ -4795,4 +4803,129 @@ some_function(
             )"""
         @test fmt(str_, 4, 1) == str
     end
+    @testset "Format docstrings" begin
+        unformatted = """
+        \"""
+        This is a docstring
+
+        ```@example
+        a =  1
+         b  = 2
+         a + b
+        ```
+
+        ```jldoctest
+        a =  1
+        b  = 2
+        a + b
+
+        # output
+
+        3
+        ```
+
+        ```jldoctest
+        julia> a =  1
+        1
+
+        julia> b  = 2;
+
+        julia>  a + b
+        3
+
+        julia> function test(x)
+               x + 1
+               x + 2
+               end;
+        ```
+        \"""
+        function test(x) x end"""
+
+        formatted = """
+        \"""
+        This is a docstring
+
+        ```@example
+        a = 1
+        b = 2
+        a + b
+        ```
+
+        ```jldoctest
+        a = 1
+        b = 2
+        a + b
+
+        # output
+
+        3
+        ```
+
+        ```jldoctest
+        julia> a = 1
+        1
+
+        julia> b = 2;
+
+
+        julia> a + b
+        3
+
+        julia> function test(x)
+                   x + 1
+                   x + 2
+               end;
+
+        ```
+        \"""
+        function test(x)
+            x
+        end"""
+        @test fmt(unformatted) == formatted
+    end
+    @testset "Multi-line indented code-blocks" begin
+        unformatted = """
+        \"""
+            format_text(
+            )
+        \"""
+        function format_text() end"""
+
+        formatted = """
+        \"""
+            format_text(
+            )
+        \"""
+        function format_text() end"""
+        @test format_text(unformatted) == formatted
+    end
+    @testset "Empty line in docstring" begin
+        unformatted = """
+        \"""
+
+        \"""
+        function test() end"""
+
+        formatted = """
+        \"""
+        \"""
+        function test() end"""
+        @test format_text(unformatted) == formatted
+    end
+    @testset "Indented docstring" begin
+        unformatted = """
+        begin
+            "Indented docstring"
+            indented_item
+        end"""
+        formatted = """
+        begin
+            \"""
+            Indented docstring
+            \"""
+            indented_item
+        end"""
+        @test format_text(unformatted) == formatted
+        @test_broken format_text(formatted) == formatted
```

Comment:
What's the output difference for this?

Reply:
It ends up being

```
"""
begin
    \"""
        Indented docstring
    \"""
    indented_item
end
"""

```

Reply:
the code blocks are indented so maybe It thinks it's that? 

btw you should use fmt instead of format_text for testing because `fmt` calls format_text twice so it ensures first format == second format